packages/purpose-process/src/services/purposeService.ts

```diff
@@ -203,6 +211,60 @@ export function purposeServiceBuilder(
       });
       await repository.createEvent(event);
     },
+    async rejectPurposeVersion({
+      purposeId,
+      versionId,
+      rejectionReason,
+      organizationId,
+      correlationId,
+    }: {
+      purposeId: PurposeId;
+      versionId: PurposeVersionId;
+      rejectionReason: string;
+      organizationId: TenantId;
+      correlationId: string;
+    }): Promise<void> {
+      logger.info(`Rejecting Version ${versionId} in Purpose ${purposeId}`);
+
+      if (!rejectionReason) {
```

Comment:
can the `rejectionReason` be null? the spec constraints + zod validator should already guarantee its presence, returning 400 otherwise

Reply:
That check also covers the empty string case. Anyways, the api spec has `minimum: 10` for the `rejectionReason`, so it shouldn't be needed. Should I remove the check entirely from the logic? In this way we would not be able to cover it in the integration tests. I would keep it, since it is a business logic constraint. What do you think?

Reply:
In my opinion we can apply the same consideration of an eservice name, we require it to be non empty, but the check is not done inside the business logic function, we trust the validator layer

Reply:
Removed